### DISCUSSION TITLE
Automated code formatting

### DISCUSSION THREAD

**Discussion by braven112** [AlaskaAirlines/WC-Generator#437] [Announcements]
## Is your feature request related to a problem? Please describe.

I've noticed that feedback on PR reviews is sometimes code formatting related and it seems like something that we could automate.

I'd like to explore using something like Prettier to handle basic code formatting with the goal to reduce or eliminate the need to correct those type of issues in a PR review.

## Describe the solution you'd like

A solution that allows people to contribute without first having to read and comprehend a large document of contribution rules.  Having a good set of contribution rules is of course really valuable, so let's automate the ones we can and consider removing rules that can't be automated or that don't bring clear value.

## Describe alternatives you've considered

Yell at people until they do the right thing. :D

## Additional context
Not sure to what extent this has already been explored but I'm assuming it has been to some extent.

**Comment by geoffrich** [comment]
I've been a fan of using Prettier in other projects. It's nice for automating away code formatting debates.

**Comment by blackfalcon** [comment]
Do you have examples of reviews that inspired this issue? I am curious as to what our current setup is not catching. I mean, we have ALL the linters and there is a [editor config](https://github.com/AlaskaAirlines/WC-Generator/blob/master/.editorconfig) file?

**Comment by braven112** [comment]
> Do you have examples of reviews that inspired this issue? I am curious as to what our current setup is not catching. I mean, we have ALL the linters and there is a [editor config](https://github.com/AlaskaAirlines/WC-Generator/blob/master/.editorconfig) file?

Yeah I was running auro-suggest the other day and it reminded me.  I think I run into this on all the repos (on windows).  My first step after opening vs code is to change all the line breaks to `lf`.  I never really understood why because all the config's say it should be `lf`.

In this case at least, I had to turn off the linters to even get the project to run. 

![image](https://user-images.githubusercontent.com/5118118/138019833-28829a89-1508-4030-a03a-345c3b415017.png)

Whereas when I use projects with prettier (that other people way smarter than me set up), like [this one](https://github.com/Alaska-ECommerce/App-Sitecore9Client) I've never had a problem.  It was setup to do formatting on save which was great and the settings seemed to actually work in vs code on windows or mac, either one, without any fancy environment setup.  It's using eslint along with prettier.

The other example that I remember from the past was an issue with the sass linter not working.  Maybe that was my problem in the case above? Prettier can format scss.

With the current linters I believe we do them as a pre-commit, right?  The downside there is when you do have issues you have to go back and fix them one by one. With prettier it would just keep it pretty in the first place.  We could still have the pre-commit linters if we still think they are valuable but at least they wouldn't fail very often.

**Comment by blackfalcon** [comment]
I am not really up to speed on the issues we have with Windows and I am not sure how using prettier will fix issues with Windows formatting or why these linters are not working. 

From what I remember, the updates you are removing were the things we did to address issues with Windows. 

@geoffrich can you help shed some light on this?

**Comment by braven112** [comment]
@Eliza-Huang @Izook @codetothepoint Have you guys ran into anything similar or have any suggestions?

**Comment by blackfalcon** [comment]
I added the `question` label to this issue. There is still a question here as to what problem we are really solving.

**Comment by Eliza-Huang** [comment]
On Windows, on some of our projects, there are red squiggly lines when there are linting errors that help indicate where and how we should fix it. There is also a `--fix` flag that we use with linting commands that will fix easy to fix linting problems for us such as periods at the end of sentences, changing `"` to `'`, and line breaks.

My experience with Auro is that instead of being notified of those errors during development, it notifies us afterwards in the commit process. If it is possible to notify us of linting errors during development, it would be good to have documentation or set up instructions for it. Additionally, a `--fix` flag would save even more time.

**Comment by blackfalcon** [comment]
Adding to this discussion, we could look at using formatting tools in the pre-commit process 

https://prettier.io/docs/en/precommit.html

There is also this issue about creating a stylelint config file 

https://github.com/AlaskaAirlines/eslint-config-auro/issues/1
